frontend/src/components/ProfileContent.tsx

```diff
@@ -50,46 +61,107 @@ const useStyles = makeStyles({
         padding: 5,
         paddingRight: 10,
     },
+    bottomRight: {
+        position: 'absolute',
+        bottom: '0',
+        right: '0',
+        transform: 'translate(-10%, -10%)',
+    },
+    bottomRight2: {
+        position: 'absolute',
+        bottom: '0',
+        right: '0',
+        transform: 'translate(-100%, -10%)',
+    },
 });
 
 export default function ProfileContent(props: ProfileContentProps) {
     const classes = useStyles();
+    const [isEditMode, toggleEditMode] = React.useState(false)
+
+    // Is used to compare fields that have been edited with their
+    // original values.
+    const [name, setName] = React.useState<string>()
+    const [username, setUsername] = React.useState<string>()
+    const [age, setAge] = React.useState<number | string>()
+    const [gender, setGender] = React.useState<string>()
+    const [occupations, setOccupations] = React.useState<string[]>()
+    const [location, setLocation] = React.useState<string>()
+    const [interests, setInterests] = React.useState<string[]>()
+    const [religion, setReligion] = React.useState<string>()
+    const [editedProfileFields, setEditableProfileFields] = React.useState<ProfileFieldList | null>(null)
+
+    const readOnlyContent = 'readonly';
+    const editableContent = 'editable';
```

Comment:
Why not just put these inline?

Reply:
I find it better practice to pull IDs out into a const to avoid typos and it makes it easier to change the ID in the future